javascript/extractor/src/com/semmle/js/parser/TypeScriptParserMetadata.java

```diff
@@ -0,0 +1,67 @@
+package com.semmle.js.parser;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+
+/**
+ * Static data from the TypeScript compiler needed for decoding ASTs.
+ * <p>
+ * AST nodes store their kind and flags as integers, but the meaning of this integer changes
+ * between compiler versions. The metadata contains mappings from integers to logical names
+ * which are stable across versions.
+ */
+public class TypeScriptParserMetadata {
+  private final JsonObject nodeFlags;
+  private final JsonObject syntaxKinds;
+  private final Map<Integer, String> nodeFlagMap = new LinkedHashMap<>();
+  private final Map<Integer, String> syntaxKindMap = new LinkedHashMap<>();
```

Comment:
How about replacing this with two instances of the Guava `HashBiMap`. 
And afterwards change the public API of `TypeScriptParserMetadata` to: 
 `getNodeFlagId(String name)`, `getNodeFlagName(int id)`, `getSyntaxKindId(String name)`, `getSyntaxKindName(int id)`.

Reply:
It's not actually a bijective map hence the condition `!idToName.containsKey(prim.getAsInt())` when we construct it.

I've come to be a bit wary of classes that shield an underlying Map by reimplementing its API, but since we're mixing Map and JsonObjects here I can see it's a bit more ergonomic here. I've implemented the second suggestion. PTAL.